hopsworks-common/src/main/java/io/hops/hopsworks/common/airflow/AirflowManager.java

```diff
@@ -132,19 +148,150 @@ public void init() throws RuntimeException {
     } catch (Exception ex) {
       throw new RuntimeException(ex);
     }
+    
+    try {
+      recover();
+    } catch (Exception ex) {
+      LOG.log(Level.WARNING, "Failed to recover material for Airflow sessions", ex);
+    }
+    
     long interval = Math.max(3000L, settings.getJWTExpLeewaySec() * 1000 / 2);
     timerService.createIntervalTimer(10L, interval, new TimerConfig("Airflow JWT renewal", false));
   }
   
+  private void recover() {
```

Comment:
@kouzant add some comments on the workflow of this method. It's not short enough to be self-explained.